mod-source-record-storage-server/src/main/java/org/folio/dao/PostgresClientFactory.java

```diff
@@ -221,16 +218,7 @@ private static PgPool getCachedPool(Vertx vertx, String tenantId) {
   }
 
   private static PgConnectOptions getConnectOptions(String tenantId) {
-    return new PgConnectOptions()
-      .setHost(postgresConfig.getString(HOST))
-      .setPort(postgresConfig.getInteger(PORT))
-      .setDatabase(postgresConfig.getString(DATABASE))
-      .setUser(postgresConfig.getString(USERNAME))
-      .setPassword(postgresConfig.getString(PASSWORD))
-      .setIdleTimeout(postgresConfig.getInteger(IDLE_TIMEOUT, 60000))
-      .setIdleTimeoutUnit(TimeUnit.MILLISECONDS)
-      .setReconnectAttempts(postgresConfig.getInteger(DB_RECONNECTATTEMPTS, 0))
-      .setReconnectInterval(postgresConfig.getLong(DB_RECONNECTINTERVAL, 1L))
+    return PgConnectOptionsHelper.createPgConnectOptions(postgresConfig)
```

Comment:
looks like we also need to provide SSL configuration for dataSource pool in case SERVER_PEM is set
https://github.com/folio-org/mod-source-record-storage/blob/267a0ff5d2a55daca86ec43ff2910d00918d90a9/mod-source-record-storage-server/src/main/java/org/folio/dao/PostgresClientFactory.java#L237